File: src/main/java/com/etour/app/controller/PassengerController.java
package com.etour.app.controller;

import com.etour.app.entity.PassengerMaster;
import com.etour.app.service.PassengerService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/passengers")
public class PassengerController {

    private final PassengerService passengerService;

    public PassengerController(PassengerService passengerService) {
        this.passengerService = passengerService;
    }

    @GetMapping("/all")
    public ResponseEntity<List<PassengerMaster>> getAllPassengers() {
        return ResponseEntity.ok(passengerService.getAllPassengers());
    }

    @PostMapping("/add")
    public ResponseEntity<PassengerMaster> addPassenger(@RequestBody PassengerMaster passenger) {
        return ResponseEntity.ok(passengerService.addPassenger(passenger));
    }

    @GetMapping("/booking/{bookingId}")
    public ResponseEntity<List<PassengerMaster>> getByBookingId(@PathVariable Integer bookingId) {
        return ResponseEntity.ok(passengerService.getPassengersByBookingId(bookingId));
    }
    
    @GetMapping("/{passengerId}")
    public PassengerMaster getPassengerById(@PathVariable int passengerId) {
        return passengerService.getPassengerById(passengerId);
    }


   

    @DeleteMapping("/{passengerId}")
    public ResponseEntity<String> deletePassenger(@PathVariable Integer passengerId) {
        passengerService.deletePassenger(passengerId);
        return ResponseEntity.ok("Passenger deleted");
    }
}

File: dotnet/backend/Controllers/PassengerController.cs
using Microsoft.AspNetCore.Mvc;
using Etour_Backend_dotnet.Models;
using Etour_Backend_dotnet.Services;

namespace Etour_Backend_dotnet.Controllers;

[ApiController]
[Route("api/passengers")]
public class PassengerController : ControllerBase
{
    private readonly IPassengerService _passengerService;

    public PassengerController(IPassengerService passengerService)
    {
        _passengerService = passengerService;
    }

    // ==========================
    // GET ALL PASSENGERS
    // ==========================
    [HttpGet]
    public async Task<IActionResult> GetAllPassengers()
    {
        var passengers = await _passengerService.GetAllPassengers();
        return Ok(passengers);
    }

    // ==========================
    // ADD PASSENGER
    // ==========================
    [HttpPost("add")]
    public async Task<IActionResult> AddPassenger([FromBody] passenger_master passenger)
    {
        var result = await _passengerService.AddPassenger(passenger);
        return Ok(result);
    }

    // ==========================
    // GET BY BOOKING ID
    // ==========================
    [HttpGet("booking/{bookingId}")]
    public async Task<IActionResult> GetByBookingId(int bookingId)
    {
        var passengers = await _passengerService.GetPassengersByBookingId(bookingId);
        return Ok(passengers);
    }

    // ==========================
    // GET BY ID
    // ==========================
    [HttpGet("{passengerId}")]
    public async Task<IActionResult> GetPassengerById(int passengerId)
    {
        try
        {
            var passenger = await _passengerService.GetPassengerById(passengerId);
            return Ok(passenger);
        }
        catch (Exception ex)
        {
            return NotFound(new { message = ex.Message });
        }
    }

    // ==========================
    // DELETE PASSENGER
    // ==========================
    [HttpDelete("{passengerId}")]
    public async Task<IActionResult> DeletePassenger(int passengerId)
    {
        try
        {
            await _passengerService.DeletePassenger(passengerId);
            return Ok(new { message = "Passenger deleted" });
        }
        catch (Exception ex)
        {
            return NotFound(new { message = ex.Message });
        }
    }
}
